lib/sequent/core/event_store.rb

```diff
@@ -108,7 +108,12 @@ def resolve_event_type(event_type)
       def publish_events(events, event_handlers)
         events.each do |event|
           event_handlers.each do |handler|
-            handler.handle_message event
+            begin
+              handler.handle_message event
+            rescue => e
+              msg = "Replay failed => aggregate: #{event.aggregate_id}, event: #{event.class}, sequence_number: #{event.sequence_number}"
+              raise e, msg, e.backtrace
+            end
           end
```

Comment:
I suggest a named Error like `ReplayFailedError` with the `EventHandler`s / `Projector`s class name, the Original exception and the full event. Than the caller can make up it's own message.